source/development/coredump.rst

```diff
@@ -0,0 +1,194 @@
+.. Copyright (C) 2024, Wazuh, Inc.
+
+.. meta::
+  :description: This section contains instructions to configure and collect coredumps for furture analisys
+
+.. _coredumps:
+
+HowTo configure Wazuh agent's OS to generate `coredumps`
+========================================================
+This section contains instructions to customize the diferent OS where Wazuh agent are installed.
+
+Introduction
+------------
+A **core dump** a.k.a **crash-dump** is a snapshot of the memory of a running process. The S.O. where the Wazuh agent resides, it can automatically create a core dump when a serious or unhandled error occurs. Sometimes a crash dump is useful for diagnosing a process that appears to be hanging; The memory dump may reveal information about the cause of the crash.
+When collecting a core dump, be sure to collect other information about the environment so that the core file can be analyzed (for example, the agent and OS version along with any other data you consider appropriate).
+
+Linux agent's OS
+----------------
+Since Linux version 2.41 onwards, a template is implemented to define the location and name of the coredumps that are generated in the system, here the reference_.
+Unlike the previous versions, which, if enabled, generated the core file next to the location of the file that caused the error.
+
+.. _reference: https://man7.org/linux/man-pages/man5/core.5.html
+
+Using `ulimit`, without `systemd`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Setting up core dump generation without using systemd involves configuring the operating system's core dump settings manually. Here's how you can do it on Linux systems:
+
+1. **Identify the Target Directory for Core Dumps**:
+    Choose a directory where core dump files will be generated. This directory should have sufficient disk space and appropriate permissions for the process generating core dumps to write to it.
+
+2. **Set the Core Dump Size Limit**:
+    If the current core dump size limit is insufficient, increase it using the ``ulimit`` command. For example, to set the core dump size limit to unlimited:
+
+.. code-block:: console
+
+    ulimit -c unlimited
+..
+
+3. **Enable Core Dump Generation**:
+    Enable core dump generation by setting the ``core_pattern`` sysctl parameter to specify the core dump file pattern and location. For example, to set the core dump file pattern to ``/var/core/core.%e.%p`` (where `%e` represents the executable name and `%p` represents the process ID):
+
+.. code-block:: console
+
+    echo "/var/core/core.%e.%p" > /proc/sys/kernel/core_pattern
+..
+
+4. **Automate Configuration (Optional)**:
+    To persist these changes across reboots, add the ``ulimit`` command and ``echo`` command setting ``core_pattern`` to a startup script or system initialization script (e.g., ``/etc/rc.local``).
```

Comment:
The verb persist is intransitive or requieres a preposition (persist in, persist with). "To preserve these changes" might be a good alternative

Reply:
done